impl/src/main/java/org/ehcache/UserManagedCacheBuilder.java

```diff
@@ -268,7 +293,43 @@ public void close() throws Exception {
     }
 
   }
-  
+
+  private void validateListenerConfig() {
+    if (!eventListenerConfigurations.isEmpty() && eventNotifier instanceof DisabledCacheEventNotificationService) {
+      if (orderedExecutor == null && unOrderedExecutor == null) {
+        throw new IllegalArgumentException("Listeners will not work unless exceutors or EventDispatcher is configured.");
+      }
+    }
+  }
+
+  private void registerListeners(Cache<K, V> cache, ServiceProvider serviceLocator, List<LifeCycled> lifeCycledList) {
+    if (!eventListenerConfigurations.isEmpty()) {
+      final CacheEventListenerProvider listenerProvider = serviceLocator.getService(CacheEventListenerProvider.class);
+      if (listenerProvider == null) {
+        throw new IllegalStateException("No CacheEventListenerProvider could be found - did you configure one?");
```

Comment:
That makes little sense then. Looking at the code around all this, I believe there is something not working that well ... While I do not want to revise all of that here, I still believe that if the user provides a `CacheEventListenerConfiguration` we should make that service available.